docs/webhook.asciidoc

```diff
@@ -0,0 +1,181 @@
+[id="{p}-webhook"]
+=== Validating webhook
+
+A validating webhook provides additional validation of Elasticsearch resources: it provides immediate feedback on the Elasticsearch manifests you submit, allowing you to catch errors right away before ECK even tries to fulfill your request.
+
+[float]
+=== Architecture
+The webhook is composed of 4 main components. Here is a brief description of each of them to understand how they interact, their naming, and how they are managed.
+
+. A `ValidatingWebhookConfiguration` object that defines the validating webhook, named `elastic-es-validation.k8s.elastic.co`. It must be created before starting the operator. The `caBundle` field can be automatically managed as part of the automatic certificate management _(see below)_.
+. A Kubernetes Service is used to expose the validating server, named `elastic-webhook-server`. It is in the same Namespace as the webhook server.
+. A webhook server that actually validates the submitted resources. In ECK it is the operator itself when it is configured with the `webhook` role. See <<{p}-operator-config,Configuring ECK>> for more information about the `operator-roles` flag.
+. A Secret containing the required certificates to secure the connection between the API server and the webhook server.
+Like the ValidatingWebhookConfiguration, it must be created before starting the operator, even if it is empty. By default its name is `elastic-webhook-server-cert`.
+The content of this Secret and the lifecycle of the certificates are automatically managed for you. You can disable this feature if you want to setup the certificates by yourself or with https://github.com/jetstack/cert-manager[cert-manager]. See an example of the latter below.
```

Comment:
```suggestion
The content of this Secret and the lifecycle of the certificates are automatically managed for you. You can disable this feature if you want to manage the certificates yourself or with https://github.com/jetstack/cert-manager[cert-manager]. See an example of the latter below.
```

Reply:
Also I don't remember if I brought this up before so sorry if I am repeating myself, but it may be worthwhile to be more explicit about what specifically we mean by managing the lifecycle.

We should also bring up here that it manages the CA bundle in the webhook config (which we reference above). All we have in the config reference:
https://github.com/elastic/cloud-on-k8s/blob/master/docs/operator-config.asciidoc
is that the flag
>Enables automatic webhook certificate management

so we should be clearer about the specifics either here or in that doc.